### PR TITLE
reproducibility PR2: add new statement creation

### PR DESCRIPTION
**Pull request by shahules786**
This PR applies the new form of statement creation introduced in #924 to answer_correctness to improve it's stability and performance. 

#931

### CODE REVIEW

**Comment by jjmachan** [comment]
add discriptions to your PRs - only for once you request review!

**Review by jjmachan — CHANGES_REQUESTED** [review]
lets get PR1 merged first before this
this feels like it has commits from both?

**Comment by shahules786** [comment]
@jjmachan you're right. PR2 is to be merged after PR1. I cut a branch from PR1 because changes to be made was depended on each other.

File: src/ragas/metrics/base.py [hunk 1]
```diff
@@ -156,3 +157,30 @@ def init(self, run_config: RunConfig):
...
+            item = inputs[0][i]
+            verdicts = [inputs[k][i][attribute] for k in range(len(inputs))]
+            verdict_counts = dict(Counter(verdicts).most_common())
+            item[attribute] = list(verdict_counts.keys())[0]
+            verdict_agg.append(item)
+
+        return verdict_agg
+
+
+ensembler = Ensember()
```
Comment (jjmachan) [inline src/ragas/metrics/base.py:186]: why is ensembler a class instead of a function?

**Review by jjmachan — APPROVED** [review]
I think the last PR neads to be rebased for this to merge?

**Pull request closed without merging**